Issue with `constexpr std::array` on Resolute Racoon

```
2026-05-03T21:34:02.0821516Z     [ RUN      ] TestGenericSystem.simple_dynamics_pos_vel_acc_control_modes_interfaces
2026-05-03T21:34:02.0821748Z     [INFO] [1777843641.383337222] [TestGenericSystem]: Loading hardware 'MockHardwareSystem' 
2026-05-03T21:34:02.0822127Z     [INFO] [1777843641.383639077] [TestGenericSystem]: Loaded hardware 'MockHardwareSystem' from plugin 'mock_components/GenericSystem'
2026-05-03T21:34:02.0822515Z     [INFO] [1777843641.383751066] [TestGenericSystem]: Initialize hardware 'MockHardwareSystem' 
2026-05-03T21:34:02.0823194Z     [WARN] [1777843641.383796731] [TestGenericSystem.hardware_component.system.MockHardwareSystem]: Executor is not available during hardware component initialization for 'MockHardwareSystem'. Skipping node creation!
2026-05-03T21:34:02.0823809Z     [INFO] [1777843641.383822279] [TestGenericSystem.hardware_component.system.MockHardwareSystem]: `status_publish_rate` is set to 0.0, hardware status publisher will not be created.
2026-05-03T21:34:02.0824105Z     [INFO] [1777843641.383969804] [TestGenericSystem]: Successful initialization of hardware 'MockHardwareSystem'
2026-05-03T21:34:02.0824336Z     [INFO] [1777843641.384254857] [resource_manager]: 'configure' hardware 'MockHardwareSystem' 
2026-05-03T21:34:02.0824601Z     [INFO] [1777843641.384294090] [resource_manager]: Successful 'configure' of hardware 'MockHardwareSystem'
2026-05-03T21:34:02.0824823Z     [INFO] [1777843641.384310170] [resource_manager]: 'activate' hardware 'MockHardwareSystem' 
2026-05-03T21:34:02.0825089Z     [INFO] [1777843641.384324126] [resource_manager]: Successful 'activate' of hardware 'MockHardwareSystem'
2026-05-03T21:34:02.0825349Z     [ERROR] [1777843641.384416799] [TestGenericSystem]: Not acceptable command interfaces combination: 
2026-05-03T21:34:02.0825422Z     Start interfaces: 
2026-05-03T21:34:02.0825483Z     [
2026-05-03T21:34:02.0825555Z       joint1/unknown
2026-05-03T21:34:02.0825627Z       joint2/acceleration
2026-05-03T21:34:02.0825688Z     ]
2026-05-03T21:34:02.0825761Z     Stop interfaces: 
2026-05-03T21:34:02.0825819Z     [
2026-05-03T21:34:02.0825879Z     ]
2026-05-03T21:34:02.0825948Z     Not existing: 
2026-05-03T21:34:02.0826010Z     [
2026-05-03T21:34:02.0826077Z      joint1/unknown
2026-05-03T21:34:02.0826136Z     ]
2026-05-03T21:34:02.0826198Z     
2026-05-03T21:34:02.0826725Z     [ERROR] [1777843641.384560238] [TestGenericSystem.hardware_component.system.MockHardwareSystem]: Joint 'joint1' has to have 'position', 'velocity', or 'acceleration' interface!
2026-05-03T21:34:02.0827104Z     [ERROR] [1777843641.384591616] [TestGenericSystem]: Component 'MockHardwareSystem' did not accept command interfaces combination: 
2026-05-03T21:34:02.0827475Z     Start interfaces: 
2026-05-03T21:34:02.0827541Z     [
2026-05-03T21:34:02.0827616Z       joint2/acceleration
2026-05-03T21:34:02.0827679Z       joint1/effort
2026-05-03T21:34:02.0827742Z     ]
2026-05-03T21:34:02.0827810Z     Stop interfaces: 
2026-05-03T21:34:02.0827867Z     [
2026-05-03T21:34:02.0828012Z     ]
2026-05-03T21:34:02.0828076Z     
2026-05-03T21:34:02.0828732Z     [ERROR] [1777843641.384653041] [TestGenericSystem.hardware_component.system.MockHardwareSystem]: Got multiple (2) starting interfaces for joint 'joint2' - this is not supported!
2026-05-03T21:34:02.0829106Z     [ERROR] [1777843641.384677727] [TestGenericSystem]: Component 'MockHardwareSystem' did not accept command interfaces combination: 
2026-05-03T21:34:02.0829181Z     Start interfaces: 
2026-05-03T21:34:02.0829238Z     [
2026-05-03T21:34:02.0829308Z       joint2/acceleration
2026-05-03T21:34:02.0829377Z       joint1/position
2026-05-03T21:34:02.0829444Z       joint2/velocity
2026-05-03T21:34:02.0829500Z     ]
2026-05-03T21:34:02.0829567Z     Stop interfaces: 
2026-05-03T21:34:02.0829628Z     [
2026-05-03T21:34:02.0829683Z     ]
2026-05-03T21:34:02.0829737Z     
2026-05-03T21:34:02.0830481Z     /usr/include/c++/15/array:210: constexpr std::array<_Tp, _Nm>::value_type& std::array<_Tp, _Nm>::operator[](size_type) [with _Tp = double; long unsigned int _Nm = 3; reference = double&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
2026-05-03T21:34:02.0830570Z     Stack trace (most recent call last):
2026-05-03T21:34:02.0830802Z     #23   Object "/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2", at 0xffffffffffffffff, in 
2026-05-03T21:34:02.0831235Z     #22   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x5590320844c4, in _start
2026-05-03T21:34:02.0831515Z     #21   Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7f97b03d1717, in __libc_start_main
2026-05-03T21:34:02.0831734Z     #20   Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7f97b03d1600, in 
2026-05-03T21:34:02.0832151Z     #19   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x5590320f10fc, in main
2026-05-03T21:34:02.0832603Z     #18   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x5590320f7ba7, in RUN_ALL_TESTS()
2026-05-03T21:34:02.0833095Z     #17   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x559032144e89, in testing::UnitTest::Run()
2026-05-03T21:34:02.0834157Z     #16   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x55903215eca6, in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)
2026-05-03T21:34:02.0834714Z     #15   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x559032154d7b, in testing::internal::UnitTestImpl::RunAllTests()
2026-05-03T21:34:02.0835201Z     #14   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x55903214fc06, in testing::TestSuite::Run()
2026-05-03T21:34:02.0835674Z     #13   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x559032144df4, in testing::TestInfo::Run()
2026-05-03T21:34:02.0836130Z     #12   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x559032144c6d, in testing::Test::Run()
2026-05-03T21:34:02.0836997Z     #11   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x55903215e646, in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)
2026-05-03T21:34:02.0838011Z     #10   Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/build/hardware_interface/test_generic_system", at 0x5590320ced73, in TestGenericSystem_simple_dynamics_pos_vel_acc_control_modes_interfaces_Test::TestBody()
2026-05-03T21:34:02.0838823Z     #9    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libhardware_interface.so", at 0x7f97b0ebb213, in hardware_interface::ResourceManager::read(rclcpp::Time const&, rclcpp::Duration const&)
2026-05-03T21:34:02.0840457Z     #8    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libhardware_interface.so", at 0x7f97b0eba3fc, in auto hardware_interface::ResourceManager::read(rclcpp::Time const&, rclcpp::Duration const&)::{lambda(auto:1&, bool)#1}::operator()<std::vector<hardware_interface::HardwareComponent, std::allocator<hardware_interface::HardwareComponent> > >(std::vector<hardware_interface::HardwareComponent, std::allocator<hardware_interface::HardwareComponent> >&, bool) const
2026-05-03T21:34:02.0841203Z     #7    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libhardware_interface.so", at 0x7f97b0f3b06b, in hardware_interface::HardwareComponent::read(rclcpp::Time const&, rclcpp::Duration const&)
2026-05-03T21:34:02.0842007Z     #6    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libhardware_interface.so", at 0x7f97b0f49e62, in hardware_interface::HardwareComponentInterface::trigger_read(rclcpp::Time const&, rclcpp::Duration const&)
2026-05-03T21:34:02.0842762Z     #5    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libmock_components.so", at 0x7f97ac3e5051, in mock_components::GenericSystem::read(rclcpp::Time const&, rclcpp::Duration const&)
2026-05-03T21:34:02.0843405Z     #4    Object "/home/runner/work/ros2_control/ros2_control/.work/target_ws/install/hardware_interface/lib/libmock_components.so", at 0x7f97ac3e95ab, in std::array<double, 3ul>::operator[](unsigned long)
2026-05-03T21:34:02.0843822Z     #3    Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.35", at 0x7f97b079c715, in std::__glibcxx_assert_fail(char const*, int, char const*, char const*)
2026-05-03T21:34:02.0844002Z     #2    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7f97b03cf8eb, in abort
2026-05-03T21:34:02.0844169Z     #1    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7f97b03ecb7d, in raise
2026-05-03T21:34:02.0844372Z     #0    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7f97b044d48c, in pthread_kill
2026-05-03T21:34:02.0844464Z     Aborted (Signal sent by tkill() 7586 0)
2026-05-03T21:34:02.0844523Z   >>>
```

**To Reproduce**
https://github.com/ros-controls/ros2_control/actions/runs/25291134652/job/74143007051?pr=3268

--- a/hardware_interface/src/mock_components/generic_system.cpp
+++ b/hardware_interface/src/mock_components/generic_system.cpp
@@ -467,7 +467,7 @@ return_type GenericSystem::read(const rclcpp::Time & /*time*/, const rclcpp::Dur
         }
       }
       // mirror them back
-      for (size_t i = 0; i < standard_interfaces_.size(); ++i)
+      for (size_t i = 0; i < joint_state_values_.size(); ++i)
       {
         if (
           std::isfinite(joint_state_values_[i]) &&

